source/slang/emit.cpp

```diff
@@ -3599,34 +3601,58 @@ struct EmitVisitor
         }
     }
 
-    void emitComparison(EmitContext* ctx, IRInst* inst, IREmitMode mode, EOpInfo& inOutOuterPrec, const EOpInfo& opPrec, bool* needCloseOut)
+    void _maybeEmitGLSLVectorCast(EmitContext* ctx, IRType* vecType, IRInst* inst, IREmitMode mode, const EOpInfo& opPrec)
     {
-        *needCloseOut = maybeEmitParens(inOutOuterPrec, opPrec);
-
-        if (getTarget(ctx) == CodeGenTarget::GLSL
-            && as<IRVectorType>(inst->getOperand(0)->getDataType())
-            && as<IRVectorType>(inst->getOperand(1)->getDataType()))
+        if (as<IRVectorType>(inst->getDataType()) == nullptr)
         {
-            const char* funcName = getGLSLVectorCompareFunctionName(inst->op);
-            SLANG_ASSERT(funcName);
-
-            emit(funcName);
+            emitIRType(ctx, vecType);
```

Comment:
Technically this path should be computing the `opPrec` for a call, and then using `maybeEmitParens` just like the other expression emit cases do.

This won't matter for now because we'd be passing in `kEOp_General` if you make that edit, and that would guarantee that we'd never need parentheses when emitting a call here.